src/main/java/org/jenkinsci/plugins/corsfilter/AccessControlsFilter.java

```diff
@@ -182,6 +183,13 @@ void reloadAllowedOriginsList() {
             return allowedOriginsList;
         }
 
+        public Object readResolve() throws ObjectStreamException {
+            if (allowedOriginsList == null) {
```

Comment:
No need to check for null, it will be always null in all correct use-cases. Better to always update it